cc.newspring.Apollos/Rest/Controllers/GiveController.cs

```diff
@@ -206,504 +274,571 @@ public HttpResponseMessage ScheduleGiving( [FromBody]ScheduleParameters schedule
-                if ( giveParameters.AccountType.ToLower() == "credit" )
-                {
-                    paymentInfo = new CreditCardPaymentInfo()
-                    {
-                        Number = giveParameters.AccountNumber,
-                        Code = giveParameters.CCV,
-                        ExpirationDate = new DateTime( giveParameters.ExpirationYear, giveParameters.ExpirationMonth, 1 ),
-                        BillingStreet1 = giveParameters.Street1,
-                        BillingStreet2 = giveParameters.Street2,
-                        BillingCity = giveParameters.City,
-                        BillingState = giveParameters.State,
-                        BillingPostalCode = giveParameters.PostalCode,
-                        BillingCountry = giveParameters.Country
-                    };
-                }
-                else
-                {
-                    paymentInfo = new ACHPaymentInfo()
-                    {
-                        BankRoutingNumber = giveParameters.RoutingNumber,
-                        BankAccountNumber = giveParameters.AccountNumber,
-                        AccountType = giveParameters.AccountType.ToLower() == "checking" ? BankAccountType.Checking : BankAccountType.Savings
-                    };
-                }
+                GenerateResponse( HttpStatusCode.BadRequest, "Schedule must contain a FrequencyValueGuid" );
+                return null;
             }
 
-            paymentInfo.Amount = totalAmount;
-            paymentInfo.FirstName = giveParameters.FirstName;
-            paymentInfo.LastName = giveParameters.LastName;
-            paymentInfo.Email = giveParameters.Email;
-            paymentInfo.Phone = giveParameters.PhoneNumber;
-            paymentInfo.Street1 = giveParameters.Street1;
-            paymentInfo.Street2 = giveParameters.Street2;
-            paymentInfo.City = giveParameters.City;
-            paymentInfo.State = giveParameters.State;
-            paymentInfo.PostalCode = giveParameters.PostalCode;
-            paymentInfo.Country = giveParameters.Country;
+            var frequency = DefinedValueCache.Read( scheduleParameters.FrequencyValueGuid.Value );
 
-            return paymentInfo;
+            if ( frequency == null )
+            {
+                GenerateResponse( HttpStatusCode.BadRequest, "Schedule must contain a valid FrequencyValueGuid" );
+                return null;
+            }
+
+            return new PaymentSchedule { 
+                PersonId = person.Id,
+                StartDate = scheduleParameters.StartDate.Value,
+                TransactionFrequencyValue = frequency
+            };
         }
 
-        private Person GetGiverPerson( GiveParameters giveParameters, RockContext rockContext, out HttpResponseMessage errorResponse )
+        private PaymentInfo GetPaymentInfo( PaymentParameters parameters, Person person, RockContext rockContext, decimal totalAmount, FinancialPaymentDetail paymentDetail )

```

Comment:
@benwiley these methods should be documented before it goes to production.  Ctrl+Shift+D as long as you have GhostDoc installed.